content/blog/nielx/2022-12-16_http_network_services_preview_in_r1_beta_4.md

```diff
@@ -0,0 +1,109 @@
+
+<!--more-->
+
+## What's included in the Preview
+
+The preview that is part of Haiku R1 Beta 4 includes a basic HTTP 1.1 library. It supports the following:
+
+* HTTP 1.1 Requests with:
+    - Standard and custom methods (GET, POST, etc.)
+    - Optional request body with a known size and content type
+    - Custom HTTP Header Fields
+    - HTTP Basic Authentication
+* HTTP 1.1 responses with:
+    - Request headers
+    - Response bodies as a string, or written to specific files
+* Configurable automatic handling of HTTP redirects
+* Configuration of the number of concurrent requests per host and in total
+
+Protocols other than HTTP 1.1, or additional functionality like proxies, cookie handling and other authentication types are not supported yet.
+
+## Getting Started with the Preview
+
+The new HTTP library was first merged in [`hrev56572`](https://git.haiku-os.org/haiku/commit/?id=1111dda699f8093bfdc9b232a919110a1c1b49a7), and is part of the Haiku R1 Beta 4 development package.
+
+There is a specific section in the [API documentation](https://www.haiku-os.org/docs/api/group__netservices.html). The headers can be found in `/boot/system/develop/headers/private/netservices2/` and in order to use the API, you should link against `libnetservices2.a`.
+
+If you use the Makefile engine to build your application, then you should tweak the `LIBS` to add the `libnetservices2.a` library, as well as the applicable C++ runtime libraries (`$(STDCPPLIBS)`) and the network libraries (`libnetwork.so` and `libbnetapi.so`). You also need to add the headers to the `SYSTEM_INCLUDE_PATHS`. For example:
+
+```
+LIBS =  be netservices2 $(STDCPPLIBS) network bnetapi
+
+SYSTEM_INCLUDE_PATHS =  /boot/system/develop/headers/be \
+ /boot/system/develop/headers/cpp \
+ /boot/system/develop/headers/private/netservices2
+```
+
+If you are building your applications on a 32 bit Haiku image, you will have to make sure you use the modern compiler kit. If you are getting unexpected/odd build errors, double check you have run `setarch x86`.
+
+In the nightly images and the master branch, development continues, and therefore there may be incompatible changes to the kit. This can either mean that your source will no longer build, or the binaries will not work correctly if the implementation of the functionality has changed. If you do decide to release software based on the preview, then make sure to build it against a known Haiku image. Because the preview functionality is statically linked in your application, it is likely any binaries will continue to work on more modern Haiku builds.
```

Comment:
"or the binaries will not work correctly if the implementation of the functionality has changed" seems a bit confusing, I guess it refers to freshly built binaries due to API changes, but it's not very clear and at first glance appears to contradict the next sentence: "it is likely any binaries will continue to work on more modern Haiku builds"